### PR TITLE
refactor CMF and add test

### PR DESCRIPTION
**Pull request by tsrnc2**
Used shawn8901s suggestion with fixes

### CODE REVIEW

File: src/strategy/indicators/cmf.spec.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { CMF } from './cmf'
+import { candles } from './spec.helper'
+
+fdescribe('CMF', () => {
+  fit('should calculate the correct cmf', () => {
+    const cmf = CMF.calculate(candles.slice(0, 19), 20)
+ //   expect(cmf).toEqual(1.099868227)
```
Comment (Shawn8901) [inline src/strategy/indicators/cmf.spec.ts]: Could you please fill the test case here? So we have an little test, which ensures that no one breaks this indicator.

File: src/strategy/indicators/index.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 export * from './ema'
 export * from './rsi'
-export * from './sma'
+export * from '/.cmf'
```
Comment (Shawn8901) [inline src/strategy/indicators/index.ts]: I would prefer a new line at end of file (like github also hints, that its missing)

File: src/strategy/indicators/cmf.spec.ts (hunk 1 above)
Comment (notVitaliy) [reply src/strategy/indicators/cmf.spec.ts]: Also, remove the `f` from `fdescribe` and `fit`

File: src/strategy/indicators/cmf.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// Chaikin Money Flow
+export class CMF {
+  public static calculate(periods: Record<string, number>[], length: number) {
+    if (periods.length >= length) {
+      let moneyFlowVolume = 0
+      let sumOfVolume = 0
+    
+      periods.slice(0,length).forEach((curr) => {
```
Comment (Shawn8901) [inline src/strategy/indicators/cmf.ts]: you could already unpack the curr within the forEach, if you do the following, which is more easy to read

>       periods.slice(0,length).forEach(({close, low, high, volume}) => {
>        moneyFlowVolume += (volume * (close - low - (high - close))) / (high - low)
 >       sumOfVolume += volume
>      })

File: src/strategy/indicators/cmf.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// Chaikin Money Flow
```
Comment (Shawn8901) [inline src/strategy/indicators/cmf.ts]: If you start the comment with `/**` and end it with `*/` afterwards (multi line comment) an ide like vscode will display the comment as class comment in the tooltip. 
i would suggest to add the comment below as the class comment

**Review by Shawn8901 — CHANGES_REQUESTED** [review]

File: src/strategy/indicators/index.ts [hunk 5]
```diff
@@ -1,3 +1,3 @@
 export * from './ema'
 export * from './rsi'
-export * from '/.cmf'
\ No newline at end of file
+export * from '/.cmf'
```
Comment (Shawn8901) [inline src/strategy/indicators/index.ts]: 2 points here. i think `./cmf` is right here and you removed `export * from './sma'` in your first commit, i think this was not   intended

**Review by Shawn8901 — CHANGES_REQUESTED** [review]

File: src/strategy/indicators/cmf.ts (hunk 3 above)
Comment (notVitaliy) [reply src/strategy/indicators/cmf.ts]: Or you could 1-line it.

```typescript
const [moneyFlowVolume, sumOfVolume] = periods.slice(0, length).reduce((([a, b], { close: c, low: l, high: h, volume: v }) => (a += (v * (c - l - (h - c))) / (h - l)), (b += v), [a, b]), [0, 0])
```

Comment (Shawn8901) [reply src/strategy/indicators/cmf.ts]: thats maybe a one liner but for me somehow hard to understand. i would prefer my code :-D

Comment (notVitaliy) [reply src/strategy/indicators/cmf.ts]: :trollface:

**Comment by tsrnc2** [comment]
I prefer to be verbose unless the one line method is the project standard

**Review by notVitaliy — APPROVED** [review]
LGTM

**Pull request merged**